src/rashdf/geom.py

```diff
@@ -364,6 +369,64 @@ def refinement_regions(self) -> GeoDataFrame:
             crs=self.projection(),
         )
 
+    def structures(self) -> GeoDataFrame:
+        """Return the model structures.
+
+        Returns
+        -------
+        GeoDataFrame
+            A GeoDataFrame containing the model structures if they exist.
+        """
+        if "/Geometry/Structures" not in self:
+            return GeoDataFrame()
+        struct_data = self["/Geometry/Structures"]
+        v_conv_val = np.vectorize(convert_ras_hdf_value)
+        struct_dict = {
+            "struct_id": range(struct_data["Attributes"][()].shape[0]),
+            "Type": v_conv_val(struct_data["Attributes"][()]["Type"]),
+            "Mode": v_conv_val(struct_data["Attributes"][()]["Mode"]),
+            "River": v_conv_val(struct_data["Attributes"][()]["River"]),
+            "Reach": v_conv_val(struct_data["Attributes"][()]["Reach"]),
+            "RS": v_conv_val(struct_data["Attributes"][()]["RS"]),
+            "Connection": v_conv_val(struct_data["Attributes"][()]["Connection"]),
+            "US Type": v_conv_val(struct_data["Attributes"][()]["US Type"]),
+            "US River": v_conv_val(struct_data["Attributes"][()]["US River"]),
+            "US Reach": v_conv_val(struct_data["Attributes"][()]["US Reach"]),
+            "US RS": v_conv_val(struct_data["Attributes"][()]["US RS"]),
+            "US SA/2D": v_conv_val(struct_data["Attributes"][()]["US SA/2D"]),
+            "DS Type": v_conv_val(struct_data["Attributes"][()]["DS Type"]),
+            "DS River": v_conv_val(struct_data["Attributes"][()]["DS River"]),
+            "DS Reach": v_conv_val(struct_data["Attributes"][()]["DS Reach"]),
+            "DS RS": v_conv_val(struct_data["Attributes"][()]["DS RS"]),
+            "DS SA/2D": v_conv_val(struct_data["Attributes"][()]["DS SA/2D"]),
```

Comment:
Seeing that there's a ton of different columns in this table, e.g.:
`US Ineff Left Elev`
`US Ineff Right Sta`
`Weir Coef`
`Weir Shape`
_many more..._

I'm not sure what many of those columns mean, but I know that some of the ones not listed in the `struct_dict` here would be useful in certain situations. What would you think about pulling out all of the columns rather than calling out specific ones like this?

Reply:
I updated to include all attribute fields in the `structures` `GeoDataFrame`.